index/scorch/scorch.go

```diff
@@ -308,11 +310,13 @@ func (s *Scorch) Batch(batch *index.Batch) (err error) {
 	s.fireEvent(EventKindBatchIntroductionStart, 0)
 
 	var newSegment segment.Segment
+	var bufBytes uint64
 	if len(analysisResults) > 0 {
-		newSegment, err = zap.AnalysisResultsToSegmentBase(analysisResults, DefaultChunkFactor)
+		newSegment, bufBytes, err = zap.AnalysisResultsToSegmentBase(analysisResults, DefaultChunkFactor)
 		if err != nil {
 			return err
 		}
+		atomic.AddUint64(&s.iStats.newSegBufBytesAdded, bufBytes)
```

Comment:
I have the feeling that this change won't do the trick...

That is, by the time the bufBytes has been added to the stats here, it's too late and the show's already over.

Reply:
I'm no longer convinced any of our memory accounting "does the trick".

Reply:
@steveyen Let me plug in some timing stats here, to get that information.